js/reference.js

```diff
@@ -4081,39 +4083,111 @@
+                     * If no constraintName is provided, uses the other foreign key found on the table as the leaf
+                     *   NOTE: when no constraintName, this is only called if there are 2 foreign keys and we know the main column
+                     *
+                     * @param {string[][] | string[]} constraintNameProp constraint name of the foreingkey from annotation or array of constraint names
+                     * @returns BulkCreateForeignKeyObject if leaf column can be found, null otherwise
+                     */
+                    var findLeafColumnAndSetBulkCreate = function(constraintNameProp) {
+                        /**
+                         *
+                         * @param {ERMRest.Column} col foreign key column to check if it's in the list of visible columns and matches the constraint name
+                         * @param {string[]} constraintName
+                         * @returns the foreign key column that represents the leaf table
+                         */
+                        var findLeafColumn = function(col, constraintName) {
+                            var foundColumn = null;
+
+                            for (k = 0; k < nonSystemColumnFks.length; k++) {
+                                var fk = nonSystemColumnFks[k];
+                                // make sure column is in visible columns and is in foreign key list
+                                // column and foreign key `.name` property is a hash value
+                                if (col.name === fk.name) {
+                                    if (constraintName && constraintName === col._constraintName) {
+                                        // use the constraint name to check each column and ensure it's the one we want from annotation
+                                        foundColumn = col;
+                                    } else if (!constraintName && col.name !== mainColumn.name) {
+                                        // make sure the current column is the NOT the main column
+                                        // assume there are only 2 foreign keys and we know mainColumn already
+                                        foundColumn = col;
+                                    }
+
+                                    if (foundColumn) break;
+                                }
+                            };
+
+                            return foundColumn;
+                        }
+
+                        var leafCol = null;
+                        // use for loop so we can break if we find the leaf column
+                        for (i = 0; i < self.columns.length; i++) {
+                            var column = self.columns[i];
+
+                            // column should be a foreignkey pseudo column
+                            if (!column.isForeignKey) continue;
```

Comment:
Does the chaise code handle composite fks for the "leaf column"? I think it handles composite fks for the "main column" but no the leaf. If that's the case you should also check for that here and don't allow data modelers to define this for a composite fk.